sway/tree/output.c

```diff
@@ -390,6 +391,34 @@ void output_get_box(struct sway_output *output, struct wlr_box *box) {
 	box->height = output->height;
 }
 
+static void handle_destroy_non_desktop(struct wl_listener *listener, void *data) {
+	struct sway_output_non_desktop *output =
+		wl_container_of(listener, output, destroy);
+
+	sway_log(SWAY_DEBUG, "Destroying non-desktop output '%s'", output->wlr_output->name);
+
+	int index = list_find(root->non_desktop_outputs, output);
+	list_del(root->non_desktop_outputs, index);
+
+	wl_list_remove(&output->destroy.link);
+
+	output->wlr_output->data = NULL;
+	output->wlr_output = NULL;
```

Comment:
You forgot to free the output here, it should replace the need for those 2 lines.

Reply:
Good catch